blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/AbstractElasticIO.java

```diff
@@ -97,50 +100,51 @@
     }
 
     /**
-     * This method returns a list of MetricToken's matching the given glob query.
+     * This method returns a list of {@link MetricName}'s matching the given glob query.
      *
-     * for metric names: foo.bar.xxx,
-     *                   foo.bar.baz.qux,
+     * for metrics: foo.bar.xxx,
+     *              foo.bar.baz.qux,
      *
-     * for query=foo.bar.*, returns the below list of metric tokens
+     * for query=foo.bar.*, returns the below list of metric names
      *
-     * new MetricToken("foo.bar.xxx", true)   <- From metric foo.bar.xxx
-     * new MetricToken("foo.bar.baz", false)  <- From metric foo.bar.baz.qux
+     * new MetricName("foo.bar.xxx", true)   <- From metric foo.bar.xxx
+     * new MetricName("foo.bar.baz", false)  <- From metric foo.bar.baz.qux
      *
      * @param tenant
      * @param query is glob representation of hierarchical levels of token. Ex: foo.bar.*
      * @return
      * @throws Exception
      */
-    public List<MetricToken> getMetricTokens(final String tenant, final String query) throws Exception {
+    public List<MetricName> getMetricNames(final String tenant, final String query) throws Exception {
 
-        Timer.Context esMetricTokensQueryTimerCtx = esMetricTokensQueryTimer.time();
+        Timer.Context esMetricNamesQueryTimerCtx = esMetricNamesQueryTimer.time();
         SearchResponse response;
 
         try {
-            response = getMetricTokensFromES(tenant, regexForPrevToNextLevel(query));
+            response = getMetricNamesFromES(tenant, regexToGrabCurrentAndNextLevel(query));
         } finally {
-            esMetricTokensQueryTimerCtx.stop();
+            esMetricNamesQueryTimerCtx.stop();
         }
 
-        int totalTokens = getTotalTokens(query);
+        // if query = foo.bar.*, base level is 3 which is equal to the number of tokens in the query.
```

Comment:
Maybe precede this comment with "In this example..".  I was a bit confused where 3 came from for a bit.